### DISCUSSION TITLE
API v2

### DISCUSSION THREAD

**Discussion by djong1** [djong1/gp3s#33] [Ideas]
We are gathering the requirements for a new external API and would like to get more input from the community. 

Our intention, with the new API is that certified (comply with privacy terms) partners can consume the gp3s data to support local initiatives or to create new apps and insights for our community.  It would be great to get your data needs gathered in this discussion.

From a privacy perspective we are a bit worried to expose public data to external parties and we are not sure if our community is in favor of this approach. It would be great to get your perspective in this discussion

From a developer perspective we are also not sure if there is a demand for this and if some serious apps will be developed. It would be great to gather this in this discussion.

**Comment by djong1** [comment]
@jhdebruin

**Comment by Logiqx** [comment]
Some items from wish list:

### Retain compatibility with the V1 API

Ideally any API enhancements should be non-breaking changes. I have code which uses Event_Rankings, Event_Sessions, Spot_Rankings and User_Sessions.


### Consistency of Event_Rankings and Event_Sessions

Event rankings are now automatic (allowing multiple events per session) but it is not clear whether the session list (Event_Sessions) uses the event mappings table or just returns sessions where the event was selected by the user.

The Event_Sessions method should return all of the sessions that considered for the rankings and not the event that the user manually selected.


### Addition of a Spot_Sessions method

It would be extremely useful to have a Spot_Sessions method on the GP3S API.

Spot_Sessions would benefit from a mandatory year parameter and optional month parameter.

Further details available in issue #21.


### Addition of Country_Rankings and Country_Sessions methods

Should speak for itself, just like Spot_Rankings and Spot_Sessions.

Country_Sessions would benefit from a mandatory year parameter and optional month parameter.


### Support for gps-foilsurfing

Full details of current behaviour available in issue #24.


### Additional speed types in Event_Sessions and User_Sessions

Whilst using the API, I noticed that a number of the speed types are absent in Event_Sessions and User_Sessions.

- speed_24hour
- speed_alpha_racing
- speed_foil
- speed_formula
- speed_halfhour

Full details in issue #16.


### Drop session_id parameter from User_Sessions

This parameter doesn't have any practical use and as far as I can tell is not implemented.

**Comment by djong1** [comment]
see update https://github.com/djong1/gp3s/issues/16#issuecomment-1218364270

### Delivered for review:

- Consistency of Event_Rankings and Event_Sessions
- Additional speed types in Event_Sessions (not all)

**Reply by Logiqx** [reply]
@djong1 

I've re-run the [reports](https://logiqx.github.io/gp3s-query/events/) using the V2 API. Inconsistencies are still present between Event_Rankings and Event_Sessions.

Additional speed types working fine.

**Reply by Logiqx** [reply]
Issue #116 may have fixed the consistency issues. Latest run of the reports is showing a clean bill of health.
